### PR TITLE
Adds the constable job

### PR DESCRIPTION
**Pull request by farie82**
## What Does This PR Do
Adds the constable job (jr. officer) to the game.
This job will be a stepping stone for players to start playing security or just for players who want to have a lower action packed round as security. They will NOT be here to hunt antags. Instead they will be here to respond to minor crimes and will be more RP focused than their action packet bigger brother the security officer.
They will have:
- Less access. They can get into security, processing and the briefing room and use the consoles/brig cells
- Less equipment. They start with a normal sec headset. Not a bowmans. Their glasses are not the sunglass type. They won't get a stun baton or taser and instead will have to rely on pepperspray, police batons, flashes and e-bolas. No gloves either so no touching evidence
- A different SOP preventing them from hunting antags. Will be written by Neca once they get the time
- Less armored clothing. Their clothing/armor is based of sec gear but the defense is slightly lowered.

There will be 3 constables at roundstart possible. I've lowered the security officer limit by 1 to give antags a bit of slack.

The brig briefing room has been changed to fit 3 lockers for the constables.

The sprites used in the PR are still drafts and are subject to change. Many thanks to @McRamon for providing them!

Other changes:
- ACCESS_SECURITY_EQUIPMENT is created 
- Gives the brig lobby room with consoles the `ACCESS_SECURITY_EQUIPMENT` access
- Makes most sec consoles use the `ACCESS_SECURITY_EQUIPMENT` access instead of the `ACCESS_SECURITY` or `ACCESS_FORENSICS_LOCKERS` access
- The sec checkpoint now is on `ACCESS_SECURITY_EQUIPMENT` access instead of `ACCESS_SECURITY`
- Beepsky now works on the new access
- The brig doc can't use the security equipment anymore such as the timers and consoles
- The classic baton now fits in security armor

Took https://www.paradisestation.org/forum/topic/18970-new-role-constable/ as basis

TODO:
New sprites for:
- A secure locker sprite
- A figurine (very optional)
- Special jumpsuit sprites for drasks etc

A better mapping solution is very welcome. (I'm shit with mapping so I wanted to make the least amount of changes)

This will not change the time requirement for regular security officers. This can be done later once the job is more integrated into the game.

## Why It's Good For The Game
Currently security has a lot to deal with in a round. From hunting antags to resolving civil disputes such as who stole somebodies sweatroll. This also makes security very rough to get started with for new players. Who are basically thrown into the dark in the hope they survive and actually help the crew instead of harm.
This job will help existing officers who don't have to deal with greytiders anymore and will give new players an easier job to start learning security. Also I myself would love to play as one and just RP instead of having to respond to 5 vampire calls at the same time.

Be sure to read the forum page as well for more info

## Images of changes
image
Brig briefing room changes

image
Starting look

image
Fully equiped look (sweater, belt, baton) With HUD icon

image
Sweater and cap variant

image
Starting equipment

image
What the locker currently has

## Changelog
:cl: Farie82 McRamon
add: The Constable job is added. A junior security officer whose job it is to deal with minor crimes and NOT hunt antags. A more RP oriented version of security officer
tweak: The brig briefing room has been changed to add 3 new lockers for the constables
tweak: The brig lobby room with consoles now has the new security equipment access requirement
tweak: The arrivals security checkpoint now has the new security equipment access requirement
tweak: Most security equipment (beepsky, consoles, brig timers) now use the new security equipment access requirement
tweak: The brig doc can't use security equipment anymore such as the brig timers and consoles
tweak: The classic baton now fits in security armor
/:cl:

### CODE REVIEW

**Comment by AffectedArc07** [comment]
image
Make sure you haven't overwritten anything else with the editor

**Comment by farie82** [comment]
> image
> Make sure you haven't overwritten anything else with the editor

Good spot. Still learning how to use the bastard

**Comment by TDSSS** [comment]
I think they should have sunglasses. One advantage of using a flash is that if you're disarmed, it won't do your attacker any good since you're immune. It would be all too easy for a greytider to just disarm a constable and then use his flash against him.

Also, what consoles in brig can they still use? It might be good to let them use some of them so they can be useful in a support role during major emergencies (like manning a camera console to give the frontline sec a heads-up)

**Comment by farie82** [comment]
> I think they should have sunglasses. One advantage of using a flash is that if you're disarmed, it won't do your attacker any good since you're immune. It would be all too easy for a greytider to just disarm a constable and then use his flash against him.

Might include that yeah will think about it. Stealing a sec HUD and getting sunglasses to make sunglassed HUDs isn't difficult either. Having them as constable also means they can use the AOE of the flash.

> Also, what consoles in brig can they still use? It might be good to let them use some of them so they can be useful in a support role during major emergencies (like manning a camera console to give the frontline sec a heads-up)

They will be able to use those consoles yeah. Camera, arrest console. Cell console. So they can be the administrative worker if they want

**Comment by AffectedArc07** [comment]
DNM since it requires map changes on meta & delta

**Comment by SteelSlayer** [comment]
I fear that the constables won't be able to take on any greytider that is wearing a mask, due to their only stunning tool being pepperspray. I think they might need a weak stunning tool, perhaps one of those wooden police batons or something.

**Comment by kenken244** [comment]
They should probably have proper handcuffs instead of zip ties. It's going to be a pain trying to get new cuffs after each arrest. Unless they have access to the sec vendor, in which case they will just get new handcuffs every round and it just adds another noob trap

**Comment by TDSSS** [comment]
Agreed on the handcuffs. Autolathes can make cuffs anyway and any greytider can just make cable cuffs, so it's not like it would be good loot worth attacking them over or anything.

**Comment by farie82** [comment]
I gave them handcuffs instead of zipties and I gave them a police baton in their locker. I'm still not 100% on that last one though but as said by Steel most tiders have a gasmask thus renders the pepperspray useless

**Comment by MRgaty** [comment]
> I gave them handcuffs instead of zipties and I gave them a police baton in their locker. I'm still not 100% on that last one though but as said by Steel most tiders have a gasmask thus renders the pepperspray useless

Isnt the batton just an item to hit people with? Dont know if beating the shit out of people is gonna be better than just a stun tbh.

**Comment by farie82** [comment]
> > I gave them handcuffs instead of zipties and I gave them a police baton in their locker. I'm still not 100% on that last one though but as said by Steel most tiders have a gasmask thus renders the pepperspray useless
> 
> Isnt the batton just an item to hit people with? Dont know if beating the shit out of people is gonna be better than just a stun tbh.

It's the same as the telebaton from heads or the cane for the NT rep. They can't collapse it so it's always in working mode and is normal sized. Other than that it's the same

**Comment by farie82** [comment]
New sprites (drafts) added. Thanks @McRamon for the work!
Do please comment on them so he can improve them if needed!

**Comment by Necaladun** [comment]
Code Green
The Same SOP as Security Officer applies, however:

1. Constables are required to state the reasons behind an arrest and announce the arrest over radio, before any further action is taken. 

4. Constables are permitted to carry a police baton, flash, pepper spray, and bolas. 

9. Constables must defer to the authority of any higher ranking officer - Pod Pilots, Officers, and obviously the Warden and HoS.

10. Constables on patrol must not enter maintenence without a higher ranking officer. 

11. Constables may only respond alone to medium or lower crimes. Violent offenders, EoCs, and other such threats should be responded to by higher ranking officers.



Code Blue
1. Constable guidelines for green apply, except:

2. Constables may be armed as a security officer, at the Warden's discretion. It is advised to only do so if an excess number of weapons are available.

5. Constables may not demand entry to specific Departments during regular patrols;

6. Constables may randomly search crewmembers, but are not allowed to apply any degree of force unless said crewmember acts overtly hostile. Crew who refuse to be searched may be stunned and cuffed for the search;


Code Red
1. Guidelines 2, 3, 4, 5, 6 and 7 are carried over from Code Blue;

7. Constables must report to the Warden or HoS for deployment, and are advised to never be without a higher ranking security officer.

**Comment by Necaladun** [comment]
Posting this here for now, will needed to be linked to the wiki when it's added - ping me for that - but nitpick it here at me too.

**Comment by McRamon** [comment]
Sprites are basically a placeholder draft waiting for pr to be fully approved

**Comment by farie82** [comment]
TODO fix icon merge issues

**Comment by Lynxless** [comment]
I feel like a role like this would cause issues for the antagonists as now you're dealing with another sect of people that will very likely respond to incidents near them like most non-sec do, but they're actually better equipped. Not to mention advising groupings would mean that lone antags are now being pushed further towards having an armory in their backpack that consists of either ranged 1-hit KOs (tasers, det gun, ebow) or to go full lethal (357, desword, scarp) to _ensure_ they have a chance at being able to operate.

Unless rules regarding alerts were changed since I last remember, red alert gets called the moment a single antag is found. So now you have a headache of having to deal with a minimum of 2+ groupings for most of the shift. Not to mention you're already competing against AI players that _usually_ will respond to most incidents with immediate door boltings.

So in all, I feel like if something like this came up you'd need to start buffing the other side to compensate for the increased threat

**Comment by farie82** [comment]
> I feel like a role like this would cause issues for the antagonists as now you're dealing with another sect of people that will very likely respond to incidents near them like most non-sec do, but they're actually better equipped. Not to mention advising groupings would mean that lone antags are now being pushed further towards having an armory in their backpack that consists of either ranged 1-hit KOs (tasers, det gun, ebow) or to go full lethal (357, desword, scarp) to _ensure_ they have a chance at being able to operate.
> 
> Unless rules regarding alerts were changed since I last remember, red alert gets called the moment a single antag is found. So now you have a headache of having to deal with a minimum of 2+ groupings for most of the shift. Not to mention you're already competing against AI players that _usually_ will respond to most incidents with immediate door boltings.
> 
> So in all, I feel like if something like this came up you'd need to start buffing the other side to compensate for the increased threat

A valid concern yeah. SoP should cover this and admin ruling on the job. I've also tried to compensate by lowering the security officer slots by 1. The objective here is to make a more RP and low action focused role for people that want to play as a security officer but don't want to hunt antags. If you want to hunt them then you can just become an officer which usually has free slots anyway.

I'll try to think of some ways to mitigate this risk though

**Comment by farie82** [comment]
The fedora's being deleted in icon diff bot is because they were dupes already. Icon merger fucked up somehow and I did not want to spend effort restoring sprite dupes

File: code/game/jobs/job/security.dm [hunk 1]
```diff
@@ -192,6 +192,43 @@
...
+	supervisors = "the head of security"
+	department_head = list("Head of Security")
+	selection_color = "#ffeeee"
+	access = list(ACCESS_SECURITY, ACCESS_SECURITY_EQUIPMENT, ACCESS_SEC_DOORS, ACCESS_BRIG, ACCESS_COURT, ACCESS_MAINT_TUNNELS, ACCESS_WEAPONS)
+	minimal_access = list(ACCESS_SECURITY_EQUIPMENT, ACCESS_SEC_DOORS, ACCESS_BRIG, ACCESS_COURT, ACCESS_MAINT_TUNNELS, ACCESS_WEAPONS)
+	minimal_player_age = 14
+	exp_requirements = 300
+	exp_type = EXP_TYPE_CREW
+	outfit = /datum/outfit/job/constable
+	hidden_from_job_prefs = TRUE
```
Comment (farie82) [inline code/game/jobs/job/security.dm:211]: TODO remove after TMs

**Comment by datlo** [comment]
I like this idea but I'm afraid this will just add effectively 2 extra officers for antags to deal with.

This job should have it clear, both IC and OOC, that you CANNOT hunt antags with it, even if you're part of security, like the brig phys. And this should be heavily enforced by admins.

**Comment by variableundefined** [comment]
This PR has been stale for a long time and no maintainer approval is forthcoming. I am closing it now.

**Pull request closed without merging**